internal/compute/digitalocean.go

```diff
@@ -346,120 +567,217 @@ func (p *DigitalOceanProvider) getSSHKeyID(ctx context.Context, keyName string)
 			return key.ID, nil
 		}
 	}
 
 	// If we get here, print available keys to help with diagnosis
 	if len(keys) > 0 {
-		log.Printf("Available SSH keys:")
+		logger.Infof("Available SSH keys:")
 		for _, key := range keys {
-			log.Printf("  - %s (ID: %d)", key.Name, key.ID)
+			logger.Infof("  - %s (ID: %d)", key.Name, key.ID)
 		}
 	}
 
 	return 0, fmt.Errorf("SSH key '%s' not found", keyName)
 }
 
-// SetClient sets the DOClient for testing purposes
-func (p *DigitalOceanProvider) SetClient(client types.DOClient) {
-	p.doClient = client
-}
+// createAndAttachVolumes creates and attaches volumes to a droplet
+func (p *DigitalOceanProvider) createAndAttachVolumes(
+	ctx context.Context,
+	dropletID int,
+	name string,
+	config types.InstanceConfig,
+) ([]string, []types.VolumeDetails, error) {
+	var volumeIDs []string
+	var volumeDetails []types.VolumeDetails
 
-// ValidateCredentials validates the DigitalOcean credentials
-func (p *DigitalOceanProvider) ValidateCredentials() error {
-	if p.doClient == nil {
-		return fmt.Errorf("client not initialized")
+	// If no volumes specified, return empty list
+	if len(config.Volumes) == 0 {
+		return volumeIDs, volumeDetails, nil
 	}
-	return nil
-}
 
-// waitForDeletion waits for a droplet to be fully deleted
-func (p *DigitalOceanProvider) waitForDeletion(ctx context.Context, name string, region string, maxRetries int, interval time.Duration) error {
-	if p.doClient == nil {
-		return fmt.Errorf("client not initialized")
-	}
+	logger.Infof("📦 Creating and attaching volumes for droplet %d", dropletID)
 
-	log.Printf("⏳ Waiting for droplet %s in region %s to be deleted...", name, region)
-	for i := 0; i < maxRetries; i++ {
-		// Try to list the droplet
-		droplets, _, err := p.doClient.Droplets().List(ctx, &godo.ListOptions{})
-		if err != nil {
-			fmt.Printf("⏳ Failed to list droplets: %v, retrying in %v (attempt %d/%d)...\n",
-				err, interval, i+1, maxRetries)
-			time.Sleep(interval)
-			continue
+	for _, vol := range config.Volumes {
+		// Generate unique volume name with random suffix
+		suffix := generateRandomSuffix()
+		volumeName := fmt.Sprintf("%s-%s", vol.Name, suffix)
+
+		// Create volume in the same region as the instance
+		logger.Infof("📦 Creating volume %s with size %dGiB in region %s", volumeName, vol.SizeGB, config.Region)
+		createRequest := &godo.VolumeCreateRequest{
+			Name:          volumeName,
+			Region:        config.Region,
+			SizeGigaBytes: int64(vol.SizeGB),
+			Description:   fmt.Sprintf("Volume for instance %s", name),
 		}
 
-		// Check if the droplet still exists in the specific region
-		found := false
-		for _, d := range droplets {
-			if d.Name == name && d.Region.Slug == region {
-				found = true
-				break
+		volume, resp, err := p.doClient.Storage().CreateVolume(ctx, createRequest)
+		if err != nil {
+			// Check if error is due to volume already existing
+			if resp != nil && resp.StatusCode == 409 {
```

Comment:
Would be nice to use some defined constant either from the http pkg or define it somewhere instead of hardcoded `409`